src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/channel_factory.h

```diff
@@ -28,7 +32,19 @@ class ChannelFactory {
                                 const std::string &src_name,
                                 const std::string &dst_name);
 
-  friend ChannelFactory& GetChannelFactory();
+#if defined(DDS_CHANNEL)
+  AbstractChannelPtr GetDefDDSChannel(const int &dds_depth,
+                                   const size_t &nbytes,
+                                   const std::string &topic_name,
+                                   const DDSTransportType &transport_type,
+                                   const DDSBackendType &dds_backend) {
```

Comment:
- remove nbytes and calculate it by metadata value.
- rename dds_depth as size, and keep `max_samples` at dds api
- change parameter order as following:
```
AbstractChannelPtr GetDefDDSChannel(
                                   const std::string &topic_name,
                                   const DDSTransportType &transport_type,
                                   const DDSBackendType &dds_backend,
                                   const int &size)
```

Reply:
yes, and I will also change GetDefDDSChannel to GetDDSChannel cause we have no def dds